node/src/accountant/scanners/mod.rs

```diff
@@ -2947,35 +3033,83 @@ mod tests {
     }
 
     #[test]
-    fn receivable_scanner_aborts_scan_if_no_payments_were_supplied() {
+    fn receivable_scanner_handles_no_new_payments_found() {
         init_test_logging();
         let test_name = "receivable_scanner_aborts_scan_if_no_payments_were_supplied";
-        let mut subject = ReceivableScannerBuilder::new().build();
+        let set_start_block_params_arc = Arc::new(Mutex::new(vec![]));
+        let new_start_block = 4321;
+        let persistent_config = PersistentConfigurationMock::new()
+            .set_start_block_params(&set_start_block_params_arc)
+            .set_start_block_result(Ok(()));
+        let mut subject = ReceivableScannerBuilder::new()
+            .persistent_configuration(persistent_config)
+            .build();
         let msg = ReceivedPayments {
             timestamp: SystemTime::now(),
             payments: vec![],
+            new_start_block,
             response_skeleton_opt: None,
         };
 
         let message_opt = subject.finish_scan(msg, &Logger::new(test_name));
 
         assert_eq!(message_opt, None);
+        let set_start_block_params = set_start_block_params_arc.lock().unwrap();
+        assert_eq!(*set_start_block_params, vec![4321]);
         TestLogHandler::new().exists_log_containing(&format!(
             "INFO: {test_name}: No new received payments were detected during the scanning process."
         ));
     }
 
+    #[test]
+    #[should_panic(expected = "Attempt to set new start block to 6709 failed due to: \
+    UninterpretableValue(\"Illiterate database manager\")")]
+    fn no_transactions_received_but_start_block_setting_fails() {
```

Comment:
I've added a comment above, where I think it should be a warn or an error log instead of a panic. In case you decide to go that way, this test will need to be changed.

Reply:
I explained to you in the comment attached to your original complain.